lblod/frontend-gelinkt-notuleren
# Hide template comments in all previews

### Overview
Hide template comments when previewing documents (but keep showing when editing/filling them in). 
Template comments are only needed to be shown when editing a document.

##### connected issues and PRs:
This PR needs #541 which updates the editor packages. Note that this PR is a PR to master, split from #541, which means it also includes the #541 changes. The changes for hiding template comments can all be found in `app/styles/project/_c-editor-preview.scss`
[GN-4441](https://binnenland.atlassian.net/browse/GN-4441)
Changes in the publishing service: https://github.com/lblod/notulen-prepublish-service/pull/105

### Setup
1. checkout https://github.com/lblod/notulen-prepublish-service/pull/105 and add it as your preimporter in e.g. docker-compose.override.yml of app-gelinkt-notuleren. See https://github.com/lblod/notulen-prepublish-service#development-and-testing
2. run frontend (ember serve --proxy ...)

### How to test/reproduce
Make sure all template comments are hidden (except for where editing the document).
To easily add them do a document, add 
```
<TemplateCommentsPlugin::Insert @controller={{this.controller}}/>
```
to an editor. 

### Challenges/uncertainties
Hiding is done via css, which is the most easy part (and quickest, as there is no document changing).
It was added to all "preview" classes, because it is unclear where they might be used and end up in. So safest is the always hide them.
